api/graphql/models/media_exif.go

```diff
@@ -12,11 +12,11 @@ type MediaEXIF struct {
 	DateShot        *time.Time
 	Exposure        *string
 	Aperture        *float64
-	Iso             *int
+	Iso             *int64
 	FocalLength     *float64
 	Flash           *string
-	Orientation     *int
-	ExposureProgram *int
+	Orientation     *int64
+	ExposureProgram *string
```

Comment:
Changing the type of `ExposureProgram` to a string causes two problems, first it would require some migration to make updating seemless. But more importantly, keeping it as an integer would make localization much easier in the future (see #243).

Do you think it's possible to get the raw integer value from `go-exif`?

Reply:
Yes, that should be doable.

Reply:
Okay, so to make Exiftool return ExposureProgram as an integer, I need to pass an additional argument "-n" to Exiftool. According to https://github.com/barasher/go-exiftool/blob/master/exiftool.go it is possible with extraInitArgs, but I can't figure out the correct syntax. Can you help me figure out the correct NewExiftool() call with "-n" as additional argument?

Reply:
Okay, I haven't tested it but from looking at the code you linked quickly I am guessing that you should do something like this:

```go
et, err := exiftool.NewExiftool(func(opts *Exiftool) error {
  opts.extraInitArgs = []string{"-n"}
  return nil
})
```

I might be wrong though :)

Reply:
That is similar to what I found out. Unfortunately, .extraInitArgs is not exported, thus it is not possible to parse additional startup arguments to exiftool.

So there are different ways forward:
1) Extract metadata as strings (this PR)
2) Fork go-exiftool, create support for parsing additional arguments to exiftool, and extract all metadata as integers. Note, this means that e.g. flash information is also returned as an integer. See more here: https://exiftool.org/TagNames/EXIF.html and https://exiftool.org/TagNames/EXIF.html#Flash So that would affect the internal exif parser as well.
3) ...

What do you think? In my opinion, I think this PR should be modified to make it an intermediate step towards 2) and accept that the extraction of metadata with exiftool is a work in progress.
I haven't looked much at the internal exif parser, but is it possible to extract e.g. the flash information as an integer?

Reply:
I think the best solution would be to open a pull request to the `go-exiftool` repo, making `extraInitArgs` public.
If they don't merge it then we'll maintain our own copy.

I think it's rather simple to extract the flash as an integer by using [internalExifParser.readIntegerTag](https://github.com/photoview/photoview/blob/a14b12b8d49abc20e4308ceb0ba92815db671701/api/scanner/exif/exif_parser_internal.go#L171) instead:

```go
flash, err := p.readIntegerTag(exifTags, exif.Flash, media)
```

As database migrations can add a lot of complexity, I think it would be best if we modified the database structure as little as possible. I think the approach with the least amount of extra work would be to first merge this when `go-exiftool` has been updated so flash and exposure is parsed properly.

Would you mind making the pull request to `go-exiftool`, or would you rather I did it?

Reply:
I forked go-exiftool yesterday and created a function for parsing additional arguments. I changed the import to my forked version. But then it didnt build... Should I manually edit the go.sum and go.mod files? I just deleted all barasher/go-exiftool references, not sure if that was my build problem.

Reply:
Hmm, maybe you have to run `go get github.com/Kjeldgaard/go-exiftool` to download it to the go cache first, then remove the original `go-exiftool` from Photoview's `go.mod` file to avoid conflicts.